2015/proposals/banaszkiewicz-piotr-amy.md

```diff
@@ -0,0 +1,214 @@
+[python-social-auth](http://python-social-auth.readthedocs.org/en/latest/).
+An alternative -
+[django-allauth](http://django-allauth.readthedocs.org/en/latest/) -
+seems to provide "local" authentication (ie. what we have now - via
+email/password) as well.
+
+Statistics (or *reports*) are not fancy - they are database queries; hopefully
+I'll be able to implement all of them in Django ORM.  In case of Report#4,5 or
+Report#6,7 or Report#9 I'll provide an UI to easily view specific
+people/workshops.
+
+I want to implement DataTables before diving into reports.  DataTables should
+help a lot when viewing existing listings in Amy, too.
+
+To implement charts I'll use Mozilla's
+[MetricsGraphics.js](http://metricsgraphicsjs.org/) as it's the most stable
+and well developed charting library for JavaScript I've seen in a while.
+
+## Schedule of Deliverables
+
+A word on my availability:
+
+In the month of June I have to prepare for exams. I'll make sure to be
+spending at least 20hrs per week on this project.
+
+My exams are not scheduled yet, but the timeslot for them is
+**June 23rd - July 7th**. In this time I won't be able to work on the project.
+However, most likely I'll be over with the exams by the end of June - in that
+case I'll resume working on the project as soon as I pass everything.
+
+Unless Developer guidelines aren't ready in time for GSoC 2015, I want to start
+with them so that I know what I can and cannot do.
+
+Additionally I want to point out that anyone can start working on issues this
+project will focus on during GSoC 2015.  Hence changes to the schedule below
+may be expected.
+
+### May 25th - June 7th
+
+A sort of "cleaning up" time period:
+
+* Write down Developer guidelines.
+* Push to close any remaining pull requests.
+* Create issues on [Amy's GitHub page](https://github.com/swcarpentry/amy/) for
+  the topics in this project that were not yet in there.
+* Triage GitHub issues (ie. create milestones and assign issues to them.)
+* Probably (as a result of finished Developer guidelines) launch Travis-CI and
+  Coveralls for Amy (see
+  [#166](https://github.com/swcarpentry/amy/issues/166)).
+
+### June 8th - June 21st
+
+* Implement DataTables.
+* Implement EventBrite integration.
+* Implement GitHub login integration.
+* Provide tests for these features.
+
+### June 22nd - July 5th
+
+I'm unavailable due to exams.
+
+### July 6th - July 19th
+
+* If Mailman3 is out: work with Software Carpentry admins to upgrade existing
+  Mailman 2.x [installation](http://lists.software-carpentry.org/)
+* Integrate with Mailman.
+* Test Mailman integration.
+
+### July 20th - August 2nd
+
+* Create reports and reports' pages.
+* Provide tests for these features.
+* Create colorful charts.
+
+### August 3rd - August 16th
+
+* Work on remaining issues for Amy like test fixtures or event enhancements.
+
+### August 17th - August 21th 19:00 UTC
+
+* Work on remaining issues for Amy like test fixtures or event enhancements.
```

Comment:
Maybe just a single August 3rd – August 21st slot, since you have the same activities in both slots?  Or are these slots something that you're not allowed to adjust?